### PR TITLE
Add module to list all snapshots for a host

### PR DESCRIPTION
**Pull request by Manisha15**

### CODE REVIEW

File: plugins/modules/snapshot_info.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: snapshot_info
+version_added: 1.0.0
```
Comment (evgeni) [inline plugins/modules/snapshot_info.py]:
```suggestion
version_added: 3.8.0
```

;)

File: plugins/module_utils/foreman_helper.py [hunk 2]
```diff
@@ -657,7 +657,8 @@ def list_resource(self, resource, search=None, params=None):
 
         if search is not None:
             params['search'] = search
-        params['per_page'] = PER_PAGE
+        if not resource == 'snapshots':
+            params['per_page'] = PER_PAGE
```
Comment (evgeni) [inline plugins/module_utils/foreman_helper.py]: Why is this needed? The API reply below in the fixtures does contain the `page` and `per_page` attributes, so I would expect the API to support paging?

https://github.com/ATIX-AG/foreman_snapshot_management/blob/master/app/controllers/api/v2/snapshots_controller.rb#L15 also mentions Pagination.

Comment (Manisha15) [reply plugins/module_utils/foreman_helper.py]: When recording and testing the fixtures, it was not able to match the query if there was pagination. It recorded the fixtures but threw error when testing the same fixtures with mismatch query due to pagination.

Comment (evgeni) [reply plugins/module_utils/foreman_helper.py]: :thinking: that's weird!

Would you mind dropping that change, re-record, and push that as a separate commit so that I can look at the diff?

Comment (evgeni) [reply plugins/module_utils/foreman_helper.py]: Hah!

I think I know what's up. Could it be that the apidoc snapshot (lol) we're using is older than https://github.com/ATIX-AG/foreman_snapshot_management/commit/7c91a1e9e7db3c34249d37543246e1ebc4acae2d?

```
% git log tests/fixtures/apidoc/snapshot.json
commit 241d49fd27e7f9926373a442768f12cbc2273e4e
Author: Manisha Singhal <manisha1595@gmail.com>
Date:   Thu Jun 27 14:57:05 2019 +0200

    Add Snapshot module
```

but the commit above is Dec 2019.

Please update the apidoc file and we're good! :)

Comment (Manisha15) [reply plugins/module_utils/foreman_helper.py]: Thanks, changing the apidoc file did the work.

File: plugins/modules/snapshot_info.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    required: true
+    type: str
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.infomodule
+'''
+
+EXAMPLES = '''
+- name: "Show all snapshots for a host"
+  theforeman.foreman.subscription_info:
```
Comment (evgeni) [inline plugins/modules/snapshot_info.py]:
```suggestion
  theforeman.foreman.snapshot_info:
```

File: plugins/modules/snapshot_info.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    host: "server.example.com"
+
+- name: "Show a snapshot"
+  theforeman.foreman.subscription_info:
```
Comment (evgeni) [inline plugins/modules/snapshot_info.py]:
```suggestion
  theforeman.foreman.snapshot_info:
```

**Review by evgeni — APPROVED** [review]
